### PR TITLE
Comment improvements for parse

### PR DESCRIPTION
**Pull request by lararennie**
to compare and contrast with parseAndKeepRawInput

---
This change is [Reviewable](https://reviewable.io/reviews/googlei18n/libphonenumber/1825)

### CODE REVIEW

File: java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java [hunk 1]
```diff
@@ -3018,6 +3018,12 @@ private boolean checkRegionForParsing(String numberToParse, String defaultRegion
    * is actually a valid number for a particular region is not performed. This can be done
    * separately with {@link #isValidNumber}.
    *
+   * <p> Note this method canonicalizes the phone number such that different representations can be
+   * easily compared, no matter what form it was originally entered in (e.g. national,
+   * international). If you want to record context about the number being parsed, such as the raw
+   * input that was entered, how the country code was derived etc. then call {@link
+   * #parseAndKeepRawInput} instead.
+   *
```
Comment (keghani) [inline java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java]: Needs the bit about `numberToParse can contain formatting...`

File: javascript/i18n/phonenumbers/phonenumberutil.js [hunk 2]
```diff
@@ -3977,6 +3977,12 @@ i18n.phonenumbers.PhoneNumberUtil.prototype.checkRegionForParsing_ = function(
  * convert any alpha characters into digits if it thinks this is a vanity number
  * of the type "1800 MICROSOFT".
  *
+ * Note this method canonicalizes the phone number such that different
+ * representations can be easily compared, no matter what form it was originally
+ * entered in (e.g. national, international). If you want to record context
+ * about the number being parsed, such as the raw input that was entered, how
+ * the country code was derived etc. then call parseAndKeepRawInput() instead.
+ *
```
Comment (keghani) [inline javascript/i18n/phonenumbers/phonenumberutil.js]: ditto

File: java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java (hunk 1 above)
Comment (lararennie) [inline java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java]: It is here, see the next line 3027. It is just here as a parameter comment, which we don't have in C++

File: javascript/i18n/phonenumbers/phonenumberutil.js (hunk 2 above)
Comment (lararennie) [inline javascript/i18n/phonenumbers/phonenumberutil.js]: Here at line 3991.

**Pull request merged**